core/field_textinput.js

```diff
@@ -456,7 +456,7 @@ Blockly.FieldTextInput.prototype.onHtmlInputKeyDown_ = function(e) {
     Blockly.WidgetDiv.hide();
     Blockly.DropDownDiv.hideWithoutAnimation();
   } else if (e.keyCode == Blockly.utils.KeyCodes.ESC) {
-    this.htmlInput_.value = this.htmlInput_.defaultValue;
+    this.setValue(this.htmlInput_.defaultValue);
```

Comment:
I think this should actually use `untypedDefaultValue` (which, admittedly, could use a better name and/or docs). `defaultValue` is the editor text, which may not be accepted by `setValue`.

Might be worth testing this fix with the note demo field if possible, because I think that one has the most complicated editor text behavior.

Otherwise, I think this is the correct way to fix the issue. LGTM!

Reply:
Good call.

The note field also tests correctly (before this PR esc would close the note picker and select the current node, not revert to the previous note).